### PR TITLE
Change haptics from touchBegin to touchEnd

### PR DESCRIPTION
**Pull request by Kureev**
Trigger haptics on touchEnd

### CODE REVIEW

File: ios/Rainbow.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -679,8 +679,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "export SENTRY_PROPERTIES=sentry.properties\nexport NODE_ARGS=--max-old-space-size=2048\nif [[ ! \"$CONFIGURATION\" = *Debug* ]]; then\n    DEST=$CONFIGURATION_BUILD_DIR/$UNLOCALIZED_RESOURCES_FOLDER_PATH\n    export EXTRA_PACKAGER_ARGS=\"--sourcemap-output $DEST/main.jsbundle.map\"\nfi\n\n../node_modules/@sentry/cli/bin/sentry-cli react-native xcode  ../node_modules/react-native/scripts/react-native-xcode.sh\n\nif [[ ! \"$CONFIGURATION\" = *Debug* ]]; then\n    cp $DEST/main.jsbundle* ../ios/\nfi\n";
-			showEnvVarsInLog = 0;
+			shellScript = "export SENTRY_PROPERTIES=sentry.properties\nexport NODE_ARGS=--max-old-space-size=2048\nif [[ \"$CONFIGURATION\" != \"Debug\" ]]; then\n    DEST=$CONFIGURATION_BUILD_DIR/$UNLOCALIZED_RESOURCES_FOLDER_PATH\n    export EXTRA_PACKAGER_ARGS=\"--sourcemap-output $DEST/main.jsbundle.map\"\n    cp $DEST/main.jsbundle* ../ios/\nfi\n\nnpx sentry-cli react-native xcode ../node_modules/react-native/scripts/react-native-xcode.sh\n";
```
Comment (brunobar79) [inline ios/Rainbow.xcodeproj/project.pbxproj]:
```suggestion
			shellScript = "export SENTRY_PROPERTIES=sentry.properties\nexport NODE_ARGS=--max-old-space-size=2048\nif [[ ! \"$CONFIGURATION\" = *Debug* ]]; then\n    DEST=$CONFIGURATION_BUILD_DIR/$UNLOCALIZED_RESOURCES_FOLDER_PATH\n    export EXTRA_PACKAGER_ARGS=\"--sourcemap-output $DEST/main.jsbundle.map\"\nfi\n\n../node_modules/@sentry/cli/bin/sentry-cli react-native xcode  ../node_modules/react-native/scripts/react-native-xcode.sh\n\nif [[ ! \"$CONFIGURATION\" = *Debug* ]]; then\n    cp $DEST/main.jsbundle* ../ios/\nfi\n";
			showEnvVarsInLog = 0;
```

File: ios/Button.swift [hunk 2]
```diff
@@ -56,18 +57,35 @@ class Button : RCTView {
   override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
     animateTapStart(
       duration: duration,
-      scale: scaleTo,
-      useHaptic: enableHapticFeedback ? hapticType: nil
+      scale: scaleTo
     )
+    
+    if self.enableHapticFeedback {
+      DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
```
Comment (brunobar79) [inline ios/Button.swift]: Can we move the 0.2 to a var at the top?
Maybe something like:
```
@objc var hapticFeedbackDelay: TimeInterval = 0.2
```

File: ios/TransactionListBaseCell.swift [hunk 3]
```diff
@@ -35,17 +36,32 @@ class TransactionListBaseCell : UITableViewCell {
       duration: duration,
       scale: scaleTo
     )
+    
+    DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
```
Comment (brunobar79) [inline ios/TransactionListBaseCell.swift]: Same as the other comment

File: ios/Rainbow.xcodeproj/project.pbxproj (hunk 1 above)
Comment (Kureev) [reply ios/Rainbow.xcodeproj/project.pbxproj]: Why?

File: ios/Button.swift (hunk 2 above)
Comment (Kureev) [reply ios/Button.swift]: I discussed it with @christianbaroni and we're about to remove this delay altogether

**Pull request merged**